### PR TITLE
feat(ecs): nvidia support to BottlerocketEcsVariant enum for gpu-accelerated tasks

### LINKED ISSUE #25980: (ecs): Add aws-ecs-1-nvidia to BottlerocketEcsVariant
### Describe the feature

Add support for `aws-ecs-1-nvidia` image types to ECS.BottlerocketEcsVariant (Bottlerocket image with NVIDIA support)

### Use Case

Need to run ECS application with GPU support

### Proposed Solution

Update enum with `aws-ecs-1-nvidia` option

### Other Information

_No response_

### Acknowledgements

- [x] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.83.1

### Environment details (OS name and version, etc.)

Mac OS Ventura

### PR DESCRIPTION
**Pull request by badmintoncryer**
This pull request introduces a new variant, AWS_ECS_1_NVIDIA, to the BottlerocketEcsVariant enum. This addition caters to the increasing demand for GPU-accelerated computing in containerized environments, particularly for tasks that require intensive computing power, such as machine learning and 3D rendering.

Closes #25980

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/aws-cdk-lib/aws-ecs/test/amis.test.ts
Comment (lpizzinidev) [inline packages/aws-cdk-lib/aws-ecs/test/amis.test.ts:22]:
```suggestion
  test.each([
    [ecs.BottlerocketEcsVariant.AWS_ECS_1, 'SsmParameterValueawsservicebottlerocketawsecs1nvidiax8664'],
    ...
  ])('BottleRocketImage with %s variant', (variant, ssmKey) => {
    // GIVEN
    const app = new cdk.App();
    const stack = new cdk.Stack(app);

    // WHEN
    new ecs.BottleRocketImage({
      variant,
    }).getImage(stack);

    /// THEN
    const assembly = app.synth();
    const parameters = assembly.getStackByName(stack.stackName).template.Parameters;
    expect(Object.entries(parameters).some(
      ([k, v]) => k.startsWith(ssmKey) && (v as any).Default.includes(`/bottlerocket/${variant}/x86_64/`),
    )).toEqual(true);
  });
```
Let's add a test for each variant.

**Review by lpizzinidev — CHANGES_REQUESTED** [review]
Thanks for the contribution 👍
I left some comments and suggestions.

File: packages/@aws-cdk-testing/framework-integ/test/aws-ecs/test/integ.cluster-bottlerocket-nvidia-ami.ts
Comment (badmintoncryer) [reply packages/@aws-cdk-testing/framework-integ/test/aws-ecs/test/integ.cluster-bottlerocket-nvidia-ami.ts]: Certainly! I'll try it later.

Comment (lpizzinidev) [reply packages/@aws-cdk-testing/framework-integ/test/aws-ecs/test/integ.cluster-bottlerocket-nvidia-ami.ts]: I was asking because from the Console you can deploy only with certain GPU-based instance types and it seemed strange that `c5.large` would deploy. Not sure if it's possible/worth validating this.

Comment (badmintoncryer) [reply packages/@aws-cdk-testing/framework-integ/test/aws-ecs/test/integ.cluster-bottlerocket-nvidia-ami.ts]: As you mentioned, this behavior does seem odd. However, since the integration test itself was successful, does this mean that deployment is possible via CloudFormation? Just to be sure, I have switched to integration testing with instances that have GPUs. Please let me know if further verification is needed.

**Comment by badmintoncryer** [comment]
@lpizzinidev Thank you for your review. 
I have incorporated the feedback you provided. I haven't delved deeply into the issue of being able to deploy to instances without GPUs. Please let me know if anything additional is required.

File: packages/@aws-cdk-testing/framework-integ/test/aws-ecs/test/integ.cluster-bottlerocket-nvidia-ami.ts
Comment (lpizzinidev) [reply packages/@aws-cdk-testing/framework-integ/test/aws-ecs/test/integ.cluster-bottlerocket-nvidia-ami.ts]: I guess that CloudFormation correctly deploys the ASG but the instances will fail to launch if the instance type is incompatible with the provided AMI.
We may want to leave it to the user to provide a valid configuration (unless maintainers think otherwise).

**Review by lpizzinidev — APPROVED** [review]

**Review by sumupitchayan — APPROVED** [review]

**Pull request merged**
